src/components/BooksContainer/AddBook.js

```diff
@@ -1,17 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../styles/addbook.css';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../../Redux/Books/BookState';
 
 export default function AddBook() {
+  const dispatch = useDispatch();
+
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const handleTitleChange = (e) => {
+    setTitle(e.target.value);
+  };
+
+  const handleAuthorChange = (e) => {
+    setAuthor(e.target.value);
+  };
+
+  const submitBookToStore = () => {
+    const newBook = {
+      genre: 'Thriller',
+      id: Math.floor(Math.random() * 100), // make sure it's unique
+      title,
+      author,
+      completionPercentage: '0%',
+      currentChapter: 'Chapter 0',
+    };
+
+    dispatch(addBook(newBook));
+  };
+
   return (
     <form action="#">
       <h2 className="adding-span">Add a new book</h2>
       <div className="adding-form d-flex">
-        <input className="book-input" placeholder="Title" />
-        <input className="book-input" placeholder="Author" />
-        <select className="selector">
-          <option selected disabled>Categories</option>
+        <input className="book-input" placeholder="Title" onChange={(e) => { handleTitleChange(e); }} />
+        <input className="book-input" placeholder="Author" onChange={(e) => handleAuthorChange(e)} />
+        <select defaultValue="Categories" className="selector">
+          <option disabled>Categories</option>
         </select>
-        <button type="button">ADD BOOK</button>
+        <button type="button" onClick={submitBookToStore}>ADD BOOK</button>
       </div>
     </form>
```

Comment:
- If a user adds data, the input has to be empty so that a user would add other information. please do it
- Validate the input so that a user couldn't add the empty information.

![user-11](https://user-images.githubusercontent.com/55635977/140053617-5636c3e4-ba52-4828-8e5a-00ed1744af36.png)